src/DIRAC/Resources/Computing/ARC6ComputingElement.py

```diff
@@ -0,0 +1,188 @@
+            j.JobStatusURL = arc.URL(str(statURL))
+            j.JobStatusInterfaceName = "org.nordugrid.ldapng"
+
+            mangURL = os.path.dirname(j.JobID)
+            j.JobManagementURL = arc.URL(str(mangURL))
+            j.JobManagementInterfaceName = "org.nordugrid.gridftpjob"
+
+            j.ServiceInformationURL = j.JobManagementURL
+            j.ServiceInformationInterfaceName = "org.nordugrid.ldapng"
+        else:
+            commonURL = "/".join(j.JobID.split("/")[0:4])
+            j.JobStatusURL = arc.URL(str(commonURL))
+            j.JobStatusInterfaceName = "org.nordugrid.arcrest"
+
+            j.JobManagementURL = arc.URL(str(commonURL))
+            j.JobManagementInterfaceName = "org.nordugrid.arcrest"
+
+            j.ServiceInformationURL = arc.URL(str(commonURL))
+            j.ServiceInformationInterfaceName = "org.nordugrid.arcrest"
+
+        j.PrepareHandler(self.usercfg)
+        return j
+
+    def submitJob(self, executableFile, proxy, numberOfJobs=1):
+        """Method to submit job"""
+
+        # Assume that the ARC queues are always of the format nordugrid-<batchSystem>-<queue>
+        # And none of our supported batch systems have a "-" in their name
+        self.arcQueue = self.queue.split("-", 2)[2]
+        result = self._prepareProxy()
+        if not result["OK"]:
+            self.log.error("ARCComputingElement: failed to set up proxy", result["Message"])
+            return result
+        self.usercfg.ProxyPath(os.environ["X509_USER_PROXY"])
+
+        self.log.verbose("Executable file path: %s" % executableFile)
+        if not os.access(executableFile, 5):
+            os.chmod(executableFile, stat.S_IRWXU | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH + stat.S_IXOTH)
+
+        batchIDList = []
+        stampDict = {}
+
+        # Creating an endpoint
+        endpoint = arc.Endpoint(self.ceHost, arc.Endpoint.COMPUTINGINFO, "org.nordugrid.ldapglue2")
+
+        # Get the ExecutionTargets of the ComputingElement (Can be REST, EMI-ES or GRIDFTP)
+        retriever = arc.ComputingServiceRetriever(self.usercfg, [endpoint])
+        retriever.wait()
+        targetsWithQueues = list(retriever.GetExecutionTargets())
+
+        # Targets also include queues
+        # To avoid losing time trying to submit to queues we cannot interact with, we only keep the interesting ones
+        targets = []
+        for target in targetsWithQueues:
+            if target.ComputingShare.Name == self.arcQueue:
+                self.log.debug(
+                    "Adding target:",
+                    "%s (%s)" % (target.ComputingEndpoint.URLString, target.ComputingEndpoint.InterfaceName),
+                )
+                targets.append(target)
+
+        # At this point, we should have GRIDFTP and AREX (EMI-ES and REST) targets related to arcQueue
+        # We intend to submit to AREX first, if it does not work, GRIDFTP is used
+        submissionWorked = False
+        for target in targets:
```

Comment:
How do you know the order in the targets, so that AREX comes first ? Is it because it is returned like that by retriever.GetExecutionTargets() ?